frontend/components.js

```diff
@@ -336,6 +332,57 @@ const TopLevelRouter = connect((state) => ({
   return h('main', {class: 'main'}, content)
 })
 
+class CopyOnClick extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {tempTitle: undefined}
+    this.fallbackCopyTextToClipboard = this.fallbackCopyTextToClipboard.bind(this)
+    this.copyTextToClipboard = this.copyTextToClipboard.bind(this)
+  }
+
+  fallbackCopyTextToClipboard() {
+    const input = document.createElement('textarea')
+    input.value = this.props.value
+    input.style.zIndex = '-1'
+    input.style.position = 'fixed'
+    input.style.top = '0'
+    input.style.left = '0'
+    document.body.appendChild(input)
+    input.focus()
+    input.select()
+    document.execCommand('copy')
+    document.body.removeChild(input)
+  }
+
+  async copyTextToClipboard() {
+    try {
+      if (navigator.clipboard) {
+        await navigator.clipboard.writeText(this.props.value)
+      } else {
+        this.fallbackCopyTextToClipboard()
+      }
+    } catch (err) {
+      console.error('Could not copy text: ', err)
+    } finally {
+      // TODO maybe show in better way, even for mobile
+      this.setState({tempTitle: 'Copied!'})
```

Comment:
It will not show this text in Safari. `onClick` the title will hide. In Chrome it appears again, but in Safari you have no idea what happened. Some sort of alert/notification with this message needs to show.

Reply:
I see the "copied!" title only when I hover again over the link. I'm using Firefox 59.0.2 and the same happens on Chrome as well. If it is the expected behavior, I think it wouldn't serve its purpose since virtually nobody would hover again over the text. Nevertheless, as I mentioned in the comment below, the wallet id should not be copyable for security reasons. Janci in his PR 38 (https://github.com/vacuumlabs/cardano/pull/38) provides a "copy tooltip" that does not dissapear, but you probably already know that.